### DISCUSSION TITLE
Does Pytorch Geometric have a method for integrating the top K neighbors?

### DISCUSSION THREAD

**Discussion by michaelma01** [pyg-team/pytorch_geometric#5019] [Q&A]
Hello all PyTorch Geometric Team and @rusty1s, 

I'm wondering whether the Pytorch Geometric has a method for extracting a partial graph based on the edge weight. Similar to [NeighborSampler](https://pytorch-geometric.readthedocs.io/en/2.0.1/modules/loader.html#torch_geometric.loader.NeighborSampler), for each node in a graph, I want to only integrate the information from its partial neighbor nodes when doing the message passing. However, unlike NeighborSampler, I don't want to randomly pick the specified number of nodes. Instead, I want to pick the top K neighbor nodes based on the edges' weights. 

For example, say I have a node `nA`, and it has five neighbor nodes `nB`, `nC`, `nD`, `nE`, `nF`. Their edge weights are separately 0.1, 0.2, 0.4, 0.5, 0.05. I want to only consider three of its neighbor nodes based on its weights. So when doing the message passing, I want the node `nA` to only integrate the information from `nC`, `nD` and `nE` because these three neighbor nodes have higher edge weight.

Does the Pytorch Geometric have such a method? Or do you have any suggestions how to modify the existing methods to achieve this goal? Thank you!

**Comment by rusty1s** [comment]
There does not exist such a functionality right now, sorry for that. There do exists plans though to add support for biased neighbor sampling which should cover this use-case as well.

**Comment by michaelma01** [comment]
I sse. Thank you @rusty1s.
